### PR TITLE
PR for code review

### PR DESCRIPTION
**Pull request by dili021**
- Each function is in it's own module.
- Babel configured to allow ES6 syntax in Jest.
- Tests for each function, in their own folder.

### CODE REVIEW

File: js/caesar.js [hunk 1]
```diff
@@ -1,19 +1,24 @@
...
+      encodedChar = String.fromCharCode(mod(coord, 26) + base);
+    } else {
+      encodedChar = character;
     }
-    crypt += char;
-  }
-  return crypt;
+    return encodedChar;
+  };
+  return [...string].map(char => encoder(char, offset)).join('');
```
Comment (uryelah) [inline js/caesar.js:21]: - Awesome use of the spread operator, great code.

File: js/test/analyze.test.js [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+
+test('It should return a resulting object', () => {
+  expect(typeof tester).toBe('object');
+});
+
+test('The object should have 4 properties', () => {
+  expect(Object.keys(tester).length).toEqual(4);
+});
+
+test('The properties should be average, min, max, and length', () => {
```
Comment (uryelah) [inline js/test/analyze.test.js:13]: - [x] You are missing the tests checking if the values of 'average', 'min', 'max', 'length' are correct.

File: js/test/calculator.test.js [hunk 3]
```diff
@@ -0,0 +1,21 @@
+import calculator from '../calculator';
+
+test('It should perform addition, subtraction, multiplication, and division', () => {
+  expect(Object.keys(calculator)).toContain('add', 'subtract', 'multiply', 'divide');
+});
+
+test('It should add together any number of passed numbers', () => {
+  expect(calculator.add(1, 2, 3, 4, 5)).toEqual(15);
```
Comment (uryelah) [inline js/test/calculator.test.js:17]: - [x] [OPTIONAL] It would be a good idea to have more than one test case. Try it with a different set of numbers and edge cases, like a parameter of the wrong type.

File: js/test/capitalize.test.js [hunk 4]
```diff
@@ -0,0 +1,13 @@
+import capitalize from '../capitalize';
+
+test('It should capitalize the passed in string', () => {
+  expect(capitalize('test')).toMatch('Test');
+});
+
+test('It should only affect the first character', () => {
+  expect(capitalize('test').slice(1)).toMatch('est');
```
Comment (uryelah) [inline js/test/capitalize.test.js:8]: - [ ] [OPTIONAL] Again, make sure to add more than one valid test case for the next project you add tests to.

**Review by uryelah — CHANGES_REQUESTED** [review]
Hi **Stefan Dili & Rajkumar Das Das**, I'm Sarah your reviewer for this PR 👩‍🚀 

Your project is really great and the way you use Javascript features is really cool and professional, the only issue is that some of the tests for the analyze method is missing.
Just make sure you are testing if the values in the returned object are ok and you should be good to go.

Rock on! 🎸 👨‍🎤 

# Required Changes 🏗️ 🐛 💡

(Check the comments below)

-----

 **Kindly fix the pointed out issues and comments below before requesting the next code review**

------------

Cheers and Happy coding!👏👏👏

Code review by **Sarah Uryelah Silva Chamorro**

(Feel free to contact me on slack by the handle of  **sarah** if you have any additional question)

**Review by Oghenebrume50 — APPROVED** [review]
### APPROVED

Hi @dili021 nice job :+1:  this looks good to me, kudos on making changes as earlier requested, please merge and move on 
**This is approved** 

**RAPHAEL NORIODE** :mask: STAY_SAFE
